### PR TITLE
fix: Replace unsafe lifecycle methods

### PR DESCRIPTION
**Pull request by rusackas**
🐛 Bug Fix

There were warnings about a couple of unsafe lifecycle methods happening. For one of them, I just renamed the method as the warning suggests to stop it from barking. For the other, I updated it to the suggested lifecycle method, and all seems to work well.

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]

**Comment by rusackas** [comment]
@mistercrunch re-requested a review here because rather than just renaming to the `UNSAFE_...` methods, I re-jiggered the LineMulti component. Everything seems to work fine as best as I can tell.

File: packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx [hunk 1]
```diff
@@ -59,16 +62,27 @@ function getJson(url) {
...
   componentDidMount() {
     this.loadData(this.props);
   }
 
-  // eslint-disable-next-line react/no-deprecated
-  componentWillReceiveProps(nextProps) {
-    this.loadData(nextProps);
+  shouldComponentUpdate(nextProps) {
+    const { rawFormData } = this.props;
+    const { rawFormData: nextrawFormData } = nextProps;
```
Comment (kristw) [inline packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx]: nit `next"R"awFormData`

Comment (rusackas) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx]: Thanks, just made that change.

**Comment by rusackas** [comment]
@kristw / @mistercrunch wondering if we can get this merged in :D

**Comment by kristw** [comment]
Please rebase

File: packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx [hunk 2]
```diff
@@ -58,16 +61,27 @@ function getJson(url) {
...
+  shouldComponentUpdate(nextProps) {
+    const { rawFormData } = this.props;
+    const { rawFormData: nextRawFormData } = nextProps;
+    const { queryData } = this.state;
+    if (!queryData || !isEqual(rawFormData, nextRawFormData)) {
+      return true;
+    }
+
+    return false;
+    // return false;
```
Comment (kristw) [inline packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx]: Please remove comment

**Review by kristw — APPROVED** [review]

File: packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx (hunk 2 above)
Comment (rusackas) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/LineMulti/LineMulti.jsx]: huh... wish I had a good explanation for that! It's gone now... let's see if it builds! :D

**Review by kristw — APPROVED** [review]

**Pull request merged**